src/pages/core/specification/instantiate2-algo.mdx

```diff
@@ -0,0 +1,42 @@
+---
+tags: ["core", "specification"]
+---
+
+# Instantiate2 algorithm
+
+With the instantiate2 algorithm you can create a contract address in a
+predictable and deterministic way. The underlying algorithm is rather simple.
+
+We use SHA-256 as the underlying hashing algorithm.
+
+You need to provide the following inputs:
+
+- Checksum: This _has_ to be a SHA-256 hash
+- Creator: This is the canonicalized address of, for example, the contract instantiating the
+  contract
+- Salt: This is some byte string allowing you to distinguish multiple
+  instantiations of the same contract by the same creator; this parameter has to
+  be under 64 bytes in length
+- Message: This is usually unused. CosmWasm sets this to an empty byte string
+
+Next you need to define a function to convert an unsigned 64-bit integer into
```

Comment:
It doesn't look like you need to define a function, rather you just need to convert the integer to bytes, right?

imho would be a bit clearer to just say "Next you need to convert an unsigned 64-bit integer..." and inline the call to `.to_be_bytes()`

Reply:
Yeah, makes sense. The main reason why I went for this was because I could have sworn there was an RFC that did the same thing.. well, doesn't matter.

Inlined it.